### PR TITLE
Added refs structure to all files, except folder

### PR DESCRIPTION
**Pull request by Majkluss**

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -686,13 +686,13 @@ components:
...
 
-    FileInfo:
+    GenericFileInfo:
       type: object
       required:
         - name
-        - path
         - ro
         - type
+        - size
```
Comment (HelclM) [inline spec/openapi.yaml:695]: Is this meant to also be required for folders? I think we talked about not doing it for them and also it is written in the description of size below, but in this definition it seems it is required. 

I'm not sure how precise we need to be with the definition, but theoretically it seems incorrect to me :innocent:

Comment (ondratu) [reply spec/openapi.yaml:695]: size have to be optional

File: spec/openapi.yaml [hunk 2]
```diff
@@ -719,9 +719,21 @@ components:
...
+        - type: object
+          properties:
+            refs:
+              type: object
+              properties:
+                download:
+                  type: string
+                  example: "/api/files/local/file.txt/raw"
+
     PrintFileInfo:
```
Comment (ondratu) [inline spec/openapi.yaml:734]: Please create some PrintFileSimple info without meta, so directory detail returns only PrintFileSimpleInfo but print file detail return extended PrintFileInfo ...

**Review by ondratu — CHANGES_REQUESTED** [review]

File: spec/openapi.yaml (hunk 1 above)
Comment (Majkluss) [reply spec/openapi.yaml]: Yes, my bad, it was meant only for files, but folder inherits from it. I'll fix it :)

File: spec/openapi.yaml [hunk 3]
```diff
@@ -832,11 +861,29 @@ components:
...
+              type: object
+              properties:
+                download:
+                  type: string
+                  example: "/api/files/local/firmware.hex/raw"
+
     FirmwareFileInfo:
       allOf:
-        - $ref: '#/components/schemas/FileInfo'
+        - $ref: '#/components/schemas/GenericFileInfo'
```
Comment (ondratu) [inline spec/openapi.yaml:878]: I think, that based on FirmwareFileInfoSimple is much better

**Review by ondratu — CHANGES_REQUESTED** [review]

**Review by HelclM — APPROVED** [review]

**Review by vorner — APPROVED** [review]

File: spec/openapi.yaml [hunk 4]
```diff
@@ -719,9 +719,43 @@ components:
...
+                  example: "/api/thumbnails/local/examples/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode.small.png"
+                thumbnail:
+                  type: string
+                  example: "/api/thumbnails/local/examples/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode.orig.png"
+
     PrintFileInfo:
+      description: Full print file info from the file's detail
       allOf:
-        - $ref: '#/components/schemas/FileInfo'
+        - $ref: '#/components/schemas/GenericFileInfo'
```
Comment (ondratu) [inline spec/openapi.yaml:757]: `PrintFileInfoBasic` is based on `GenericFileInfo` yet.

File: spec/openapi.yaml [hunk 5]
```diff
@@ -819,22 +853,25 @@ components:
...
-                  type: string
-                  example: "/api/thumbnails/local/examples/Spice_Harvester_0.3mm_PLA_MK3S_42m.gcode.orig.png"
+                  example: "/api/files/local/firmware.hex/raw"
 
     FirmwareFileInfo:
+      description: Full firmware file info from the file's detail
       allOf:
-        - $ref: '#/components/schemas/FileInfo'
+        - $ref: '#/components/schemas/GenericFileInfo'
+        - $ref: '#/components/schemas/FirmwareFileInfoBasic'
```
Comment (ondratu) [inline spec/openapi.yaml:874]: `FirmwareFileInfoBasic` is based on `GenericFileInfo` yet.

**Review by ondratu — CHANGES_REQUESTED** [review]

**Review by ondratu — APPROVED** [review]

**Pull request merged**